kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/VirtualKafkaClusterReconciler.java

```diff
@@ -153,6 +155,24 @@ private VirtualKafkaCluster checkClusterIngressTlsSettings(VirtualKafkaCluster c
                 return certRefCheck.get();
             }
         }
+
+        var trustRefs = cluster.getSpec().getIngresses().stream()
+                .flatMap(ingress -> Optional.ofNullable(ingress.getTls()).stream())
+                .map(Tls::getTrustAnchorRef)
+                .filter(Objects::nonNull)
+                .toList();
+        if (!trustRefs.isEmpty()) {
+            var trustAnchorCheck = trustRefs.stream()
+                    .map(trustAnchorRef -> {
+                        var path = "spec.ingresses[].tls.trustAnchor";
+                        return ResourcesUtil.checkTrustAnchorRef(cluster, context, CONFIGMAPS_EVENT_SOURCE_NAME, trustAnchorRef, path, statusFactory);
+                    })
+                    .filter(Objects::nonNull)
+                    .findFirst();
+            if (trustAnchorCheck.isPresent()) {
+                return trustAnchorCheck.get();
+            }
```

Comment:
This method is quite long now. Can we factor out a method for checking the `certRefs` and one for checking the `trustRefs`?